Source/GSXib5KeyedUnarchiver.m

```diff
@@ -0,0 +1,3391 @@
+}
+
+- (id) decodeCellsForElement: (GSXibElement*)element
+{
+  id    object = nil;
+  Class class  = NSClassFromString([element attributeForKey: @"class"]);
+
+  if ([class isSubclassOfClass: [NSMatrix class]])
+    object = [self decodeFormCellsForElement: element];
+  else
+    object = [self decodeObjectForKey: @"cells"];
+
+  return object;
+}
+
+- (id) decodeNoAutoenablesItemsForElement: (GSXibElement*)element
+{
+  NSString  *autoenablesItems = [element attributeForKey: @"autoenablesItems"];
+  BOOL       value            = NO; // Default if not present...
+
+  if (autoenablesItems)
+    value = ![autoenablesItems boolValue];
+
+  return [NSNumber numberWithBool: value];
+}
+
+- (id) decodeMenuItemForElement: (GSXibElement*)element
+{
+  NSString      *itemID   = [element attributeForKey: @"selectedItem"];
+  GSXibElement  *itemElem = [objects objectForKey: itemID];
+
+  return [self objectForXib: itemElem];
+}
+
+- (id) decodeTitleCellForElement: (GSXibElement*)element
+{
+  NSString *title   = [element attributeForKey: @"title"];
+
+  if (title)
+    {
+      id   object  = [[NSCell alloc] initTextCell: title];
+      NSFont *font = [self decodeObjectForKey: @"titleFont"];
+
+      // IF no font...
+      if (font == nil) // default to system-11...
+        font = [NSFont systemFontOfSize: 11];
+
+      [object setAlignment: NSCenterTextAlignment];
+      [object setBordered: NO];
+      [object setEditable: NO];
+      [object setFont: font];
+      return AUTORELEASE(object);
+    }
+
+  return nil;
+}
+
+- (id) decodeBorderTypeForElement: (GSXibElement*)element
+{
+  NSString      *borderType = [element attributeForKey: @"borderType"];
+  NSBorderType   value      = NSGrooveBorder; // Cocoa default...
+
+  if (borderType)
+    {
+      if ([@"bezel" isEqualToString: borderType])
+        value = NSBezelBorder;
+      else if ([@"line" isEqualToString: borderType])
+        value = NSLineBorder;
+      else if ([@"none" isEqualToString: borderType])
+        value = NSNoBorder;
+      else
+        NSWarnMLog(@"unknown border type: %@", borderType);
+    }
+
+  return [NSNumber numberWithUnsignedInteger: value];
+}
+
+- (id) decodeModifierMaskForElement: (GSXibElement*)element
+{
+  id            object     = nil;
+  NSDictionary *attributes = [[element elementForKey: @"keyEquivalentModifierMask"] attributes];
+
+  // ??? SKIP modifier mask processing if BASE64-UTF8 string being used ???
+  if (attributes == nil)
+    {
+      if (([element elementForKey: @"keyEquivalent"]) &&
+          ([[element elementForKey: @"keyEquivalent"] attributeForKey: @"base64-UTF8"]))
+      {
+        object = [NSNumber numberWithUnsignedInt: 0];
+      }
+    else
+      {
+        // Seems that Apple decided to omit this attribute IF certain default keys alone
+        // are applied.  If this key is present WITH NO setting then the following is
+        // used for the modifier mask...
+        object = [NSNumber numberWithU
```

Comment:
Urrrrrrr, how about `mask |= 1 << 24` and similar? (I hope I counted this right)

Reply:
I do not understand you point here. What do you want to change and why?

Reply:
I don't understand why Github is obnoxiously showing this comment on the wrong line.

It's referring to this code section:

```
- (id) decodeBrowserFlagsForElement: (GSXibElement*)element
{
  NSUInteger    mask                         = 0;
  NSDictionary *attributes                   = [element attributes];
  id            takesTitleFromPreviousColumn = [attributes objectForKey: @"takesTitleFromPreviousColumn"];
  id            allowsEmptySelection         = [attributes objectForKey: @"allowsEmptySelection"];
  id            acceptsArrowKeys             = [attributes objectForKey: @"acceptsArrowKeys"];

  // Set the flags...
  if ([[attributes objectForKey: @"hasHorizontalScroller"] boolValue])
    mask |= 0x10000;
  if ((allowsEmptySelection == nil) || ([allowsEmptySelection boolValue] == NO))
    mask |= 0x20000;
  if ([[attributes objectForKey: @"sendsActionOnArrowKeys"] boolValue])
    mask |= 0x40000;
  if ((acceptsArrowKeys == nil) || [acceptsArrowKeys boolValue])
    mask |= 0x100000;
  if ([[attributes objectForKey: @"separatesColumns"] boolValue])
    mask |= 0x4000000;
  if ((takesTitleFromPreviousColumn == nil) || [takesTitleFromPreviousColumn boolValue])
    mask |= 0x8000000; // Cocoa default is YES if Omitted...
  if ([[attributes objectForKey: @"titled"] boolValue])
    mask |= 0x10000000;
  if ([[attributes objectForKey: @"reusesColumns"] boolValue])
    mask |= 0x20000000;
  if ([[attributes objectForKey: @"allowsBranchSelection"] boolValue])
    mask |= 0x40000000;
  if ([[attributes objectForKey: @"allowsMultipleSelection"] boolValue])
    mask |= 0x80000000;
  if ([[attributes objectForKey: @"prefersAllColumnUserResizing"] boolValue])
    mask |= 0; // FIXME: do we handle this yet???

  return [NSNumber numberWithUnsignedInt: mask];
}
```

![image](https://user-images.githubusercontent.com/477596/71645840-475b2300-2cd6-11ea-87a4-f24e6e3d6cf2.png)

As you can see, this is different than how the PR UI is showing this:

![image](https://user-images.githubusercontent.com/477596/71645848-5a6df300-2cd6-11ea-97c5-709f05bce08e.png)

I don't know what Github did here.

Reply:
Ah now I see. Yes, this is bad code but just making the numbers more pleasant doesn't seem to be the right solution. We should define another structure to go into the GSCodingFlags.h file. But not tonight, I hope to find time for that sometimes next week.